subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/swift/SwiftPlatformToolProvider.java

```diff
@@ -129,6 +129,11 @@ private CommandLineToolInvocationWorker commandLineTool(ToolType key, String exe
     private CommandLineToolContext context(CommandLineToolConfigurationInternal toolConfiguration) {
         MutableCommandLineToolContext baseInvocation = new DefaultMutableCommandLineToolContext();
         baseInvocation.setArgAction(toolConfiguration.getArgAction());
+
+        String developerDir = System.getenv("DEVELOPER_DIR");
```

Comment:
Can this be derived from the tool installation directory?

e.g., instead of having to make sure the DEVELOPER_DIR environment variable is set properly, could we derive what DEVELOPER_DIR should be or just add something to SwiftcToolChain to set it when we configure the swift tool chain from AvailableToolChains?

It seems like that might make some of this simpler since we don't have to pass around things through environment variables.

Reply:
I don't know enough about the worker API and how they manage the environment variable. Do you know if the variables are passed to the workers?